### PR TITLE
OH2-227 refactor Story frame and content

### PR DESCRIPTION
**Pull request by ronsandova**
- [x] Create endpoint for update frame for all stories 
- [x] Add validation for frame ( member needs to own the story )
- [x] Create endpoint for update content for guided and free form
- [x] Add validation for content ( member needs to own the story )
- [x] Create endpoint for get story detail

### CODE REVIEW

File: apps/story/src/service/tag/story.tag.service.ts [hunk 1]
```diff
@@ -34,15 +31,14 @@ export class StoryTagService {
...
       tagStory.tag_id = tag.id;
-      this.logger.log('Tags ID to Story -> ' + JSON.stringify(tag.id, null, 2));
+      this.logger.log('tagStory -> ' + JSON.stringify(tagStory, null, 2));
       tags_story.push( tagStory );
     });
 
     try {
       const tags_story_entity = this.storyTagRepository.create(tags_story);
-      this.logger.log('ssss Adding Tags to Story -> ' + JSON.stringify(tags_story_entity));
+      //this.logger.log('Adding New Tags to Story -> ' + JSON.stringify(tags_story_entity));
```
Comment (jokekerker) [inline apps/story/src/service/tag/story.tag.service.ts:40]: please remove if you don't use

File: apps/story/src/service/story/story.content.service.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  private readonly logger = new Logger(StoryContentService.name);
+
+  constructor(
+    @InjectRepository(StoryEntity, Database.PRIMARY)
+    private storyRepository: Repository<StoryEntity>) {
+  }
+
+  public async updateTextFreeContent(member_id: string,
+                                     story_id: string,
+                                     dto: StoryUpdateTextFreeFormDto): Promise<IServiceResponse<UpdateResult | null>> {
```
Comment (jokekerker) [inline apps/story/src/service/story/story.content.service.ts:24]: do we allow to return null?

File: apps/story/src/saga/story.frame.update.saga.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  ) { }
+
+  /**
+   * @remarks
+   * @param StoryUpdateVideoDto | StoryUpdateTextFreeFormDto | StoryUpdateTextGuidedDto
+   * @return IServiceResponse<UpdateResult>
+   */
+
+  public async updateFrameStory( member_id: string, story_id: string, updateDto: StoryFrameUpdateDto ):
+     Promise<IServiceResponse<UpdateResult | null>> {
```
Comment (jokekerker) [inline apps/story/src/saga/story.frame.update.saga.ts:31]: return null?

File: apps/story/src/dto/story/story.update.text-guided.dto.ts [hunk 4]
```diff
@@ -1,10 +1,9 @@
-import { IsEnum, IsNotEmpty, IsOptional, IsString, MaxLength, maxLength, MinLength, minLength } from "class-validator";
+import { IsEnum, IsNotEmpty, IsOptional, IsString, MaxLength, MinLength } from "class-validator";
 import { Transform } from "class-transformer";
 import { StoryMedium, StoryType } from "../../constant/story.enum";
 import { ApiProperty } from "@nestjs/swagger";
-import { StoryUpdateBaseDto } from "./story.update.base.dto";
 
-export class StoryUpdateTextGuidedDto extends StoryUpdateBaseDto {
```
Comment (jokekerker) [inline apps/story/src/dto/story/story.update.text-guided.dto.ts:7]: if we gonna use this StoryUpdateBaseDto for contain only statis value I think just move 
these 
TEXT_MAX_LENGTH into constant folder it gonna be better

File: apps/story/src/dto/story/story.frame.update.dto.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  })
+  @MaxLength( StoryUpdateBaseDto.TEXT_MAX_LENGTH, {
+    message: 'Title cannot be longer than 100 characters'
+  })
+  @IsString()
+  public title: string;
+
+  @IsBoolean()
+  @IsOptional()
+  public has_hero_statement: boolean = false;
```
Comment (jokekerker) [inline apps/story/src/dto/story/story.frame.update.dto.ts:24]: this one is in StoryUpdateBaseDto but if we don't use it anymore please remove this StoryUpdateBaseDto class

File: apps/story/src/constant/story-patterns.constants.ts [hunk 6]
```diff
@@ -9,6 +9,11 @@ export const STORY_MESSAGE_PATTERNS = {
   UPDATE_TEXT_FREE_FORM: 'story.update-text-freeform',
   UPDATE_TEXT_GUIDE: 'story.update-text-guide',
   UPDATE_VIDEO: 'story.update-video',
```
Comment (jokekerker) [inline apps/story/src/constant/story-patterns.constants.ts:11]: we don't have any controller user with these message patterns please remove it from the gateway too.

**Review by jokekerker — CHANGES_REQUESTED** [review]

File: apps/story/src/service/story/story.content.service.ts (hunk 2 above)
Comment (ronsandova) [reply apps/story/src/service/story/story.content.service.ts:24]: we are not allow but it will return 500 in this case

**Review by jokekerker — APPROVED** [review]

**Pull request merged**